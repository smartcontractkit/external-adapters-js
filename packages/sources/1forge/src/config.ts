import { Requester } from '@chainlink/ea-bootstrap'
import { Config } from '@chainlink/types'

export const NAME = '1FORGE'

export const DEFAULT_ENDPOINT = 'forex'
export const DEFAULT_BASE_URL = 'https://api.1forge.com/'

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix, true)
  config.api = {
    ...config.api,
    baseURL: config.api.baseURL || DEFAULT_BASE_URL,
    params: {
      api_key: config.apiKey,
    },
  }
  config.defaultEndpoint = DEFAULT_ENDPOINT
  return config
}
