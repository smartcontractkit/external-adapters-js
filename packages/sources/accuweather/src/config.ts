import { Requester } from '@chainlink/ea-bootstrap'
import { Config } from '@chainlink/types'

export const NAME = 'ACCUWEATHER'
export const DEFAULT_BASE_URL = 'http://api.accuweather.com/'
export const DEV_BASE_URL = 'http://apidev.accuweather.com/'

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix)
  config.api.baseURL = config.api.baseURL || DEFAULT_BASE_URL

  return config
}
