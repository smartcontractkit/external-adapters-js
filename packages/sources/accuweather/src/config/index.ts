import { Requester } from '@chainlink/ea-bootstrap'
import { DefaultConfig } from '@chainlink/ea-bootstrap'

export const NAME = 'ACCUWEATHER'
export const DEFAULT_BASE_URL = 'http://api.accuweather.com/'
export const DEV_BASE_URL = 'http://apidev.accuweather.com/'

export const makeConfig = (prefix?: string): DefaultConfig => {
  const config = Requester.getDefaultConfig(prefix, true)
  config.api.baseURL = config.api.baseURL || DEFAULT_BASE_URL

  return config
}
