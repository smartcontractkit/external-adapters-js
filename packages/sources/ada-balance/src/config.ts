import { Requester, util } from '@chainlink/ea-bootstrap'
import { Config as BaseConfig } from '@chainlink/types'

export const NAME = 'ADA_BALANCE'

export const DEFAULT_ENDPOINT = 'balance'
export const DEFAULT_PORT = 1337

export interface ExtendedConfig extends BaseConfig {
  httpOgmiosURL?: string
  wsOgmiosURL?: string
  isTLSEnabled?: boolean
  host?: string
  port: number
}

export const makeConfig = (prefix?: string): ExtendedConfig => {
  const baseConfig = Requester.getDefaultConfig(prefix)
  baseConfig.defaultEndpoint = DEFAULT_ENDPOINT
  const rpcPort = util.getEnv('RPC_PORT', prefix)
  return {
    ...baseConfig,
    httpOgmiosURL: util.getEnv('HTTP_OGMIOS_URL', prefix),
    wsOgmiosURL: util.getEnv('WS_OGMIOS_URL', prefix),
    host: util.getEnv('WS_API_ENDPOINT', prefix),
    port: rpcPort ? parseInt(rpcPort) : DEFAULT_PORT,
    isTLSEnabled: !!util.getEnv('IS_TLS_ENABLED', prefix),
  }
}
