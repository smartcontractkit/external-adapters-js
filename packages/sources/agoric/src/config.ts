import { Requester } from '@chainlink/external-adapter'
import { Config } from '@chainlink/types'
import { util } from '@chainlink/ea-bootstrap'

export const DEFAULT_API_ENDPOINT = 'http://localhost:8000/api/oracle'

// This environment variable is needed for the Hack the Orb oracle
// instructions to remain correct.
const LEGACY_API_ENDPOINT_ENV = 'AG_SOLO_ORACLE_URL'

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix)
  config.api.baseURL =
    config.api.baseURL || util.getEnv(LEGACY_API_ENDPOINT_ENV) || DEFAULT_API_ENDPOINT
  config.apiKey = config.apiKey || 'not required'
  return config
}
