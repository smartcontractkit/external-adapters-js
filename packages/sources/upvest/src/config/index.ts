import { Requester } from '@chainlink/ea-bootstrap'
import { Config } from '@chainlink/types'

export const NAME = 'UPVEST'

export const DEFAULT_ENDPOINT = 'gasprice'
export const DEFAULT_BASE_URL = 'https://fees.upvest.co'

export const makeConfig = (prefix?: string): Config => {
  const config = Requester.getDefaultConfig(prefix)
  config.api.baseURL = config.api.baseURL || DEFAULT_BASE_URL
  config.defaultEndpoint = DEFAULT_ENDPOINT
  return config
}
