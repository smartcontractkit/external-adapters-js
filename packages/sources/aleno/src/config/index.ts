import { AdapterConfig } from '@chainlink/external-adapter-framework/config'

export const config = new AdapterConfig({
  API_KEY: {
    description: 'An API key for Aleno',
    type: 'string',
    required: true,
    sensitive: true,
  },
  WS_API_ENDPOINT: {
    description: 'WS endpoint for Aleno',
    type: 'string',
    default: 'https://state-price-socket.aleno.ai',
  },

  BACKGROUND_EXECUTE_MS: {
    description:
      'The amount of time the background execute should sleep before performing the next request',
    type: 'number',
    default: 10_000,
  },
})
