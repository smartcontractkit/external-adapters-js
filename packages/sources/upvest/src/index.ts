import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute, endpointSelector } from './adapter'
import { makeConfig, NAME } from './config'

export = {
  NAME,
  makeExecute,
  makeConfig,
  ...expose(NAME, makeExecute(), undefined, endpointSelector),
}
