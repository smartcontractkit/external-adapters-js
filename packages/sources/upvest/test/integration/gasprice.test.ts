import { AdapterRequest } from '@chainlink/ea-bootstrap'
import process from 'process'
import { server as startServer } from '../../src'
import { mockUSCPIResponseSuccess } from './fixtures'
import { DEFAULT_BASE_URL } from '../../src/config'
import { setupExternalAdapterTest } from '@chainlink/ea-test-helpers'
import type { SuiteContext } from '@chainlink/ea-test-helpers'
import { SuperTest, Test } from 'supertest'

describe('execute', () => {
  const context: SuiteContext = {
    req: null,
    server: startServer,
  }

  const envVariables = {
    API_ENDPOINT: process.env.API_ENDPOINT || DEFAULT_BASE_URL,
    API_VERBOSE: 'true',
  }

  setupExternalAdapterTest(envVariables, context)

  describe('with token', () => {
    const data: AdapterRequest = {
      id: '1',
      data: {},
    }

    it('should return success', async () => {
      mockUSCPIResponseSuccess()

      const response = await (context.req as SuperTest<Test>)
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })
})
