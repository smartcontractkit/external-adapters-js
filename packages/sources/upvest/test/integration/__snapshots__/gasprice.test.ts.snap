// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute with token should return success 1`] = `
{
  "data": {
    "estimates": {
      "fast": 131.363,
      "fastest": 132.055,
      "medium": 113.447,
      "slow": 110.747,
    },
    "result": 131.363,
    "success": true,
    "updated": "2021-11-30T15:46:00.048Z",
  },
  "jobRunID": "1",
  "providerStatusCode": 200,
  "result": 131.363,
  "statusCode": 200,
}
`;
